services/src/main/scala/cromwell/pipeline/service/ProjectFileService.scala

```diff
@@ -32,11 +33,20 @@ class ProjectFileService(
 
   def buildConfiguration(
     projectId: ProjectId,
-    projectFile: ProjectFile
-  ): Future[Either[ValidationError, ProjectConfiguration]] =
-    Future(womTool.inputsToList(projectFile.content.content)).map {
-      case Left(value) => Left(ValidationError(value.toList))
-      case Right(nodes) =>
-        Right(ProjectConfiguration(projectId, active = true, List(ProjectFileConfiguration(projectFile.path, nodes))))
+    projectFilePath: Path,
+    version: Option[PipelineVersion],
+    userId: UserId
+  ): Future[Either[Exception, ProjectConfiguration]] =
+    for {
+      project <- projectService.getUserProjectById(projectId, userId)
+      eitherFile <- projectVersioning.getFile(project, projectFilePath, version)
+    } yield eitherFile match {
+      case Right(file) =>
+        womTool.inputsToList(file.content.content) match {
+          case Right(nodes) =>
+            Right(ProjectConfiguration(projectId, active = true, List(ProjectFileConfiguration(file.path, nodes))))
+          case Left(e) => Left(ValidationError(e.toList))
```

Comment:
Is it really a validation error? Previously it was a `VersioningException.FileException`.
Why did you changed error? What this error should even mean?

Reply:
womTool returns NonEmptyList of validation errors, now I concatenate all errors to one string and throw them as ValidationExceprion.

VersioningException was my mistake.